rtengine/rawimagesource.cc

```diff
@@ -6221,12 +6227,32 @@ void RawImageSource::ItcWB(bool extra, double &tempref, double &greenref, double
         }
 
         if (oldsampling == false) {
+            if (settings->verbose) {
+                printf("size rgb loc - bfh=%i bfw=%i repref=%i\n", bfh, bfw, repref);
+            }
+            
 #ifdef _OPENMP
             #pragma omp parallel for
 #endif
 
             for (int y = 0; y < bfh ; ++y) {
                 for (int x = 0; x < bfw ; ++x) {
+                    if(!std::isnan(redloc[y][x])) {//eliminate wrong values and values not very usable for white balance
+                        clipitc(redloc[y][x]);
+                    } else {
+                        redloc[y][x] = 0.1f;
+                    }
```

Comment:
Since there is a nan check for every `clipitc`, it is cleaner to put the nan logic in `clipitc` like so:
```c++
constexpr float clipitc(float x)
{
    if (std::isnan(x)) {
        return 0.1f;
    }
    return rtengine::LIM(x, 0.1f, 65534.9f);//White balance Itcwb - limit values 
}
```

Then this piece of code can be
```suggestion
                    redloc[y][x] = clipitc(redloc[y][x]);
```
By the way, the current code does not clip the values because there is no assignment.

Reply:
This code
constexpr float clipitc(float x)
{
    if (std::isnan(x)) {
        return 0.1f;
    }
    return rtengine::LIM(x, 0.1f, 65534.9f);//White balance Itcwb - limit values 
}

 lead to
D:/code/repo-rtdr/rtengine/rawimagesource.cc:68:1: error: body of 'constexpr' function 'constexpr float {anonymous}::clipitc(float)' not a return-statement
   68 | }
      | ^
ninja: build stopped: subcommand failed.

Reply:
`constexpr` is the problem.